### PR TITLE
Add doc comment for tokenLocationPtr

### PR DESCRIPTION
**Pull request by winksaville**
The algorithm seemed unusual and I had spent some effort understanding
it, so I thought I'd add a comment.

### CODE REVIEW

**Comment by winksaville** [comment]
As this is a comment change either a different change caused it to fail or flaky Appveyor.

File: std/zig/ast.zig [hunk 1]
```diff
@@ -45,6 +45,7 @@ pub const Tree = struct {
         line_end: usize,
     };
 
+    /// Return the Location of the token relative to a line that starts at self.source[start_index].
```
Comment (andrewrk) [inline std/zig/ast.zig]: This is not quite right. It returns the location of the token relative to the offset specified by `start_index`.

**Comment by andrewrk** [comment]
The appveyor failure is #1363

**Comment by winksaville** [comment]
Txs for the correction, I said "line" because the Location returned is line oriented.

**Pull request merged**